### ISSUE TITLE
Proposal: Detect names in inline-links

### ISSUE THREAD

**Issue by lorenzleutgeb** [DavidAnson/markdownlint#213]
Suppose I want to reference/link something by name, but am doing it wrong, because I forget that I should use `[`/`]` braces instead of `(`/`)` braces:

```markdown
See [this](name).

[name]: https://example.com/name
```

Where the correct version is:

```markdown
See [this][name].

[name]: https://example.com/name
```

and the diff between wrong and correct version is

```diff
1c1
< See [this](name).
---
> See [this][name].
```

---

Markdownlint should index all names of references, and then, for all inline links, check that they are not referring to a name. In the above case I would like to see a warning along the lines of:

```
example.md: 1: MDXXX/inline-names Inline link uses reference name as URL. Don't forget to use `[...]` instead of `(...)` with named references.
```
